src/app/shared/wysiwyg/prosemirror.markdown.editor.ts

```diff
@@ -55,63 +60,67 @@ export class ProseMirrorMarkdownEditor {
     if (content !== this.content || images) {
       this.content = content;
       this.images = images || this.images;
-      let state = EditorState.create(this.stateConfig());
-      this.view.update(this.viewProps(state));
+      this.view.update(this.viewProps(EditorState.create(this.stateConfig())));
     }
   }
 
   getContent() {
     if (this.outputFormat === ProseMirrorFormatTypes.HTML) {
-      return this.removeHTML(this.view.editor.docView.dom.innerHTML);
+      return this.removeHTML(this.view.dom.innerHTML);
     } else {
-      return defaultMarkdownSerializer.serialize(this.view.editor.state.doc);
+      return defaultMarkdownSerializer.serialize(this.view.state.doc);
     }
   }
 
   destroy() {
-    this.view.editor.destroy();
+    this.view.destroy();
   }
 
   viewProps(state: any) {
-    let props = {
+    return {
       state,
+      ...(!this.editable && { editable: () => false }),
       dispatchTransaction: (transaction) => {
-        this.view.updateState(this.view.editor.state.apply(transaction));
+        this.view.updateState(this.view.state.apply(transaction));
         if (this.outputFormat === ProseMirrorFormatTypes.HTML) {
-          let newContent = this.removeHTML(this.view.editor.docView.dom.innerHTML);
+          const newContent = this.removeHTML(this.view.dom.innerHTML);
           if (this.content !== newContent) {
             this.content = newContent;
             this.setModel(this.content);
           }
         } else {
-          this.content = defaultMarkdownSerializer.serialize(this.view.editor.state.doc);
+          this.content = defaultMarkdownSerializer.serialize(this.view.state.doc);
           this.setModel(this.content);
         }
       }
     };
-
-    if (!this.editable) {
-      props['editable'] = () => false;
-    }
-
-    return props;
   }
 
   isSelectionEmpty() {
-    return this.view.editor.state.selection.empty;
+    return this.view.state.selection.empty;
   }
 
   stateConfig() {
-    let config = {plugins: this.buildMenuItemsPlugin()};
+    let doc: Node;
     if (this.inputFormat === ProseMirrorFormatTypes.HTML) {
       // use browser to create element for parsing HTML content
-      let domNode = document.createElement('div');
+      const domNode = document.createElement('div');
       domNode.innerHTML = this.content;
-      config['doc'] = DOMParser.fromSchema(basicSchema).parse(domNode);
+      doc = DOMParser.fromSchema(basicSchema).parse(domNode);
     } else {
-      config['doc'] = defaultMarkdownParser.parse(this.content ? this.content : '');
-    }
-    return config;
+      doc = defaultMarkdownParser.parse(this.content ? this.content : '');
+    }
+    return {
+      doc,
+      plugins: [
```

Comment:
* inputRules: the types of marks accepted by the schema. [This editor supports both markdown (description) and html (itunes summary) output.]
* keymap: shortcut key mappings
* menuBar: plugin containing menu items and dropdown menus
* history: \*new\* adds cmd+z undo/redo support
* class attribute for prosemirror CSS

Reply:
Nice!  Seems like some decent improvements.